opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -827,11 +827,44 @@ namespace Opm
         // only use inner well iterations for the first newton iterations.
         const int iteration_idx = simulator.model().newtonMethod().numIterations();
         if (iteration_idx < this->param_.max_niter_inner_well_iter_ || this->well_ecl_.isMultiSegment()) {
+            const auto& ws = well_state.well(this->indexOfWell());
+            const auto pmode_orig = ws.production_cmode;
+            const auto imode_orig = ws.injection_cmode;
+
+            Scalar qtotal_orig = 0.0;
+            const int np = well_state.numPhases();
```

Comment:
This code can be written the following, 

```c++
const bool all_zero_rate_original  = std::all_of(ws.surface_rates.begin(), ws.surface_rates.begin() + well_state.numPhases(), [](Scalar rate) {
    return rate == Scalar(0.0);
});

We use `all_zero_rate_original` instead of `qtotal_orig == 0.` later. I think it will be easier to read. 

It needs the header file <algorithm>